dask/dataframe/utils.py

```diff
@@ -544,6 +543,14 @@ def assert_eq_dtypes(a, b):
         if isinstance(b, (pd.DataFrame, pd.Series)):
             b = to_pyarrow_string(b)
 
+    return a, b
+
+
+def assert_eq_dtypes(a, b):
+    (
+        a,
+        b,
+    ) = _maybe_convert_string(a, b)
```

Comment:
I've not checked locally -- is `pre-commit` happy with this formatting? Not a big deal if we need what's current in this PR, but just thought I'd check

```suggestion
    a, b = _maybe_convert_string(a, b)
```

Reply:
This is what pre-commit did, for whatever reason! I didn't do that. :)

Reply:
Hrm, weird. When I make this change, `pre-commit` seems to be happy locally

Reply:
Yeah, it's happy now.